catroid/src/androidTest/java/org/catrobat/catroid/test/cloudmessaging/CloudMessagingTest.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2017 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.test.cloudmessaging;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.preference.PreferenceManager;
+import android.test.AndroidTestCase;
+
+import org.catrobat.catroid.cloudmessaging.CloudMessaging;
+import org.catrobat.catroid.ui.SettingsActivity;
+import org.junit.After;
+import org.junit.Before;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class CloudMessagingTest extends AndroidTestCase {
+
+	private Context context;
+	private SharedPreferences sharedPreferences;
+
+	@Before
+	public void setUp() {
+		sharedPreferences = mock(SharedPreferences.class);
+		context = mock(Context.class);
+		when(PreferenceManager.getDefaultSharedPreferences(context)).thenReturn(sharedPreferences);
```

Comment:
![image](https://user-images.githubusercontent.com/1794599/28877745-908dd9e6-779d-11e7-9335-d26f5bbd56be.png)

It… really… works!

k, I first thought maybe mocks don't work but stubs will. So I quickly tested it
```java
assertTrue(Foo.get());
when(Foo.get()).thenReturn(false);
assertFalse(Foo.get());
```
→ Mockito Exception :confused:

The reason why this works is because `PreferenceManager.getDefaultSharedPreferences` internally calls `context.getSharedPreferences` and context is mocked. So Mockito is smart enough to handle this case :smile:

That's great! This code is much cleaner! I like that a lot! Good work!

Maybe we can do some mocking in the CrashReporterTest as well instead of using the real SharedPreferences? However, it may need a fake (partially functional) SharedPreferences object.